src/lib.rs

```diff
@@ -1643,3 +1643,42 @@ impl std::io::Seek for OutputReader {
         Ok(self.position())
     }
 }
+
+#[cfg(feature = "rand")]
+impl rand_core::SeedableRng for OutputReader {
+    type Seed = [u8; 32];
+
+    #[inline]
+    fn from_seed(seed: Self::Seed) -> Self {
+        let mut hasher = Hasher::new();
+        hasher.update(&seed);
+        hasher.finalize_xof()
+    }
+}
+
+#[cfg(feature = "rand")]
+impl rand_core::RngCore for OutputReader {
+    #[inline]
+    fn next_u32(&mut self) -> u32 {
+        rand_core::impls::next_u32_via_fill(self)
```

Comment:
Hmmm, this is going to be somewhat inefficient, because OutputReader doesn't buffer partially consumed output blocks. Each block is 64 bytes, and we're only using 4 bytes here, so if you wind up looping on this method then your RNG throughput will be 1/16th what it should be.

We could add a 64-byte buffer to OutputReader, but that would only be a partial fix. Eventually (soon I swear!) I'm going to land the refactoring that extends SIMD optimizations to cover extended output. At that point the buffer size we need for optimal throughput will be even larger, and it'll vary by platform. I'm kind of torn about whether we should do this buffering in OutputReader itself. Thoughts?

Reply:
Hm, well, I do think that buffering inside OutputReader could prevent performance footguns like this, but it implements Read already, so most users who are using std could just wrap it in a BufferedReader for the same effect. Personally, I prefer modular solutions to loading a type with functionality it doesn't need. OutputReader already is documented to point out that it is not buffering, a note that I missed.

An easy fix for this case is to implement [BlockRngCore instead and offer a newtype wrapper around one of the BlockRng structures](https://docs.rs/rand_core/latest/rand_core/block/index.html). I'll poke at that later today.

Reply:
Oh interesting. I'm still trying to wrap my head around how Rand does things, where there are multiple different traits you can implement to become an RNG, and you only need to pick one?

Generating a block of `u32` doesn't seem too bad to me. We could either use a bit of unsafe code to reinterpret the array as `&mut [u8]` to fill it, or pull in the `bytemuck` crate to do the same thing. That said, I don't love this behavior:

> If the requested byte length is not a multiple of 4, some bytes will be discarded.

On the one hand, probably no one expects reading 1 byte at a time to be the most efficient way to do anything. On the other hand, there's no _good reason_ we need to waste bytes here. Maybe a custom struct would be better after all?

Reply:
Generally, RngCore is the important bit.  `BlockRngCore` lets it be used to be wrapped by `BlockRng` to auto-implement `RngCore`.  `ReseedingRng` also depends on `BlockRngCore`, but it all revolves around `u32` and `u64` chunks for some reason.  I really don't get that aspect, to be honest.  It seems like generating random bytes is the obvious baseline, but maybe I'm missing something.